File: ECommerceClient/ECommerceClient/src/app/shared/models/order.ts
import { DeliveryMethod } from "./deliveryMethod"

export interface Order {
  id: string
  orderDate: string
  buyerEmail: string
  shippingAddress: ShippingAddress
  shippingPrice: number
  deliveryMethod: DeliveryMethod
  paymentSummary: PaymentSummary
  orderItems: OrderItem[]
  subtotal: number
  status: string
  total: number
  paymentIntentId: string
}

export interface ShippingAddress {
  name: string
  line1: string
  line2?: string
  city: string
  state: string
  country: string
  postalCode: string
}

export interface PaymentSummary {
  last4: number
  brand: string
  expMonth: number
  expYear: number
}

export interface OrderItem {
  itemOrdered: ItemOrdered
  price: number
  quantity: number
  id: string
  createdDate: string
  updatedDate: string
}

export interface ItemOrdered {
  productId: string
  productName: string
  pictureUrl: string
}

export interface OrderToCreate{
    cartId : string
    shippingAddress: ShippingAddress
    deliveryMethodId: string
    paymentSummary: PaymentSummary
}

File: ECommerceAPI.Domain/Entities/Order/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ECommerceAPI.Domain.Entities.Order
{
    public class Order : BaseEntity
    {
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public required string BuyerEmail { get; set; }
        public ShippingAddress ShippingAddress { get; set; } = null!;
        public DeliveryMethod DeliveryMethod { get; set; } = null!;

        public PaymentSummary PaymentSummary { get; set; }
        public List<OrderItem> OrderItems { get; set; } = [];

        public decimal Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Beklemede;

        public required string PaymentIntentId { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }

    }
}
